prefect_gcp/cloud_storage.py

```diff
@@ -665,3 +665,391 @@ async def write_path(self, path: str, content: bytes) -> str:
+        """
+        if to_folder is None:
+            to_folder = ""
+        to_folder = Path(to_folder)
+
+        blobs = await self.list_blobs(folder=from_folder)
+        if len(blobs) == 0:
+            self.logger.warning(
+                f"No blobs were downloaded from "
+                f"bucket {self.bucket!r} path {from_folder!r}."
+            )
+            return to_folder
+
+        # do not call self._join_bucket_folder for list_blobs
+        # because it's built-in to that method already!
+        # however, we still need to do it because we're using relative_to
+        bucket_folder = self._join_bucket_folder(from_folder)
+        for blob in blobs:
+            bucket_path = Path(blob.name).relative_to(bucket_folder)
+            if bucket_path.is_dir():
+                continue
+            to_path = to_folder / bucket_path
+            to_path.parent.mkdir(parents=True, exist_ok=True)
+            self.logger.info(
+                f"Downloading blob from bucket {self.bucket!r} path {bucket_path!r}"
+                f"to {to_path}."
+            )
+            await run_sync_in_worker_thread(
+                blob.download_to_filename, filename=str(to_path), **download_kwargs
+            )
+        return to_folder
+
+    @sync_compatible
+    async def upload_from_path(
+        self,
+        from_path: Union[str, Path],
+        to_path: Optional[str] = None,
+        **upload_kwargs: Dict[str, Any],
+    ) -> str:
+        """
+        Uploads an object from a path to the object storage service.
+
+        Args:
+            from_path: The path to the file to upload from.
+            to_path: The path to upload the file to. If not provided, will use
+                the file name of from_path; this gets prefixed
+                with the bucket_folder.
+            **upload_kwargs: Additional keyword arguments to pass to
+                `Blob.upload_from_filename`.
+
+        Returns:
+            The path that the object was uploaded to.
+
+        Examples:
+            Upload my_folder/notes.txt object to notes.txt.
+            ```python
+            from prefect_gcp.cloud_storage import GcsBucket
+
+            gcs_bucket = GcsBucket.load("my-bucket")
+            gcs_bucket.upload_from_path("notes.txt", "my_folder/notes.txt")
+            ```
+        """
+        if to_path is None:
+            to_path = Path(from_path).name
+
+        bucket_path = self._join_bucket_folder(to_path)
+        bucket = await self.get_bucket()
+        blob = bucket.blob(bucket_path)
+        self.logger.info(
+            f"Uploading from {from_path!r} to the bucket "
+            f"{self.bucket!r} path {bucket_path!r}."
+        )
+
+        await run_sync_in_worker_thread(
+            blob.upload_from_filename, filename=from_path, **upload_kwargs
+        )
+        return bucket_path
+
+    @sync_compatible
+    async def upload_from_file_object(
+        self, from_file_object: BinaryIO, to_path: str, **upload_kwargs
```

Comment:
Do we need ignore files?

Reply:
I don't think so. That was mainly for deployments and these methods aren't going to be use for deployment storage.